### DISCUSSION TITLE
parallel processing of data

### DISCUSSION THREAD

**Discussion by wdduncan** [EvgSkv/logica#306] [General]
Is there a way to leverage parallelism in Logica? Often I use parallelism by breaking the data into groups (using group by) and then processing each group on different cores. Can I do something similar in Logica?

This discussion may be related to https://github.com/EvgSkv/logica/discussions/305.

**Comment by EvgSkv** [comment]
Okay, basically we are thinking of integrating with DuckDB. This would solve this and #305 at the same time.

Now you can load data to BigQuery and BQ will parallelize across hundreds or thousands of machines, but I am guessing you are looking for a more light weight solution.

As I mentioned in #305 DuckDB is expected to come no later than June, hoping for sooner.

**Reply by wdduncan** [reply]
Thanks. Looking forward to the new features!

**Reply by EvgSkv** [reply]
DuckDB support was added to Logica. You can use it with `@Engine("duckdb");` imperative. 
DuckDB parallelizes processing automatically, so whatever query you run, it should be using as many cores as useful. I did some quick checks and it indeed loads a lot of cores. 
Let me know if you hit any issues!

**Reply by wdduncan** [reply]
Thanks @EvgSkv !
